docs/en-us/development/development-environment-setup.md

```diff
@@ -0,0 +1,148 @@
+```
+    git clone https://github.com/apache/incubator-dolphinscheduler.git`
+```
+4. After finished the clone, go into the project folder and execute the following commands:
+```
+    git branch -a  #check the branch
+    git checkout dev #shift the dev branch
+    git pull #sychronize the branch
+    mvn -U clean package -Prelease -Dmaven.test.skip=true  #need to compile the necessary classes due to the project use gRPC dependencies
+```
+#### Install node
+
+1. Install nvm  
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh 
+
+2. Refresh the environment variables  
+source ~/.bash_profile
+
+3. Install node  
+nvm install v12.12.0  
+note:mac users could install npm through brew:brew install npm
+
+4. Validate the node installation  
+node --version
+
+#### Install zookeeper
+
+1. Download zookeeper  
+http://apache.mirrors.hoobly.com/zookeeper/zookeeper-3.6.1/apache-zookeeper-3.6.1-bin.tar.gz
+
+2. Copy the zookeeper config file  
+cp conf/zoo_sample.cfg conf/zoo.cfg
+
+3. Modify zookepper cofig  
+vi conf/zoo.cfg  
+dataDir=./tmp/zookeeper
+
+4. Start/stop zookeeper  
+./bin/zkServer.sh start  
+./bin/zkServer.sh stop
+
+#### Create database
+
+Create user, user name: ds_user, password: dolphinscheduler
+```
+    mysql> CREATE DATABASE dolphinscheduler DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_general_ci;
+    mysql> GRANT ALL PRIVILEGES ON dolphinscheduler.* TO 'ds_user'@'%' IDENTIFIED BY 'dolphinscheduler';
+    mysql> GRANT ALL PRIVILEGES ON dolphinscheduler.* TO 'ds_user'@'localhost' IDENTIFIED BY 'dolphinscheduler';
+    mysql> flush privileges;
+```
+
+#### Set up the front-end
+
+1. Enter the dolphinscheduler-ui folder  
+cd dolphinscheduler-ui
+
+2. Run command  
+npm install
+
+#### Set up the back-end
+
+1. Import the project to IDEA  
+file-->open
+
+2. Modify the database configurations, the 'datasource.properties' under the resource folder which belongs to the dao module
+```
+    spring.datasource.driver-class-name=com.mysql.jdbc.Driver
+    spring.datasource.url=jdbc:mysql://localhost:3306/dolphinscheduler
+    spring.datasource.username=ds_user
+    spring.datasource.password=dolphinscheduler  
+```
+
+3. Modify the pom.xml under the root directory, change the scope of the mysql-connector-java to complie
+
+4. Refresh the dao module  
+try to run the main method inside org.apache.dolphinscheduler.dao.upgrade.shell.CreateDolphinScheduler to insert tables and data needed for the project.
+
+5. Modify the service module  
+try to change the zookeeper.quorum part of the zookeeper.properties file
+zookeeper.quorum=localhost:2181
+
+6. Modify the dolphinscheduler-ui module  
+try to change the .env file
+```
+    API_BASE = http://localhost:12345
+    DEV_HOST = localhost
+```
+
+#### Start the project
+
+1. Start zookeeper  
+./bin/zkServer.sh start
+
+2. Start MasterServer  
+try to run the main method inside org.apache.dolphinscheduler.server.master.MasterServer and VM Options is needed:
```

Comment:
2. Start MasterServer
run the main method of org.apache.dolphinscheduler.server.master.MasterServer, you need to set the following VM options:

Reply:
done changes